modules/percolator/src/test/java/org/elasticsearch/percolator/QueryBuilderStoreTests.java

```diff
@@ -61,19 +61,17 @@ public void testStoringQueryBuilders() throws IOException {
             TermQueryBuilder[] queryBuilders = new TermQueryBuilder[randomIntBetween(1, 16)];
             IndexWriterConfig config = new IndexWriterConfig(new WhitespaceAnalyzer());
             config.setMergePolicy(NoMergePolicy.INSTANCE);
-            Settings settings = Settings.builder().put(IndexMetadata.SETTING_VERSION_CREATED, Version.CURRENT).build();
             BinaryFieldMapper fieldMapper = PercolatorFieldMapper.Builder.createQueryBuilderFieldBuilder(new ContentPath(0));
 
             Version version = Version.CURRENT;
             try (IndexWriter indexWriter = new IndexWriter(directory, config)) {
                 for (int i = 0; i < queryBuilders.length; i++) {
                     queryBuilders[i] = new TermQueryBuilder(randomAlphaOfLength(4), randomAlphaOfLength(8));
-                    ParseContext parseContext = mock(ParseContext.class);
-                    LuceneDocument document = new LuceneDocument();
-                    when(parseContext.doc()).thenReturn(document);
+                    LuceneDocument doc = new LuceneDocument();
+                    ParseContext parseContext = new TestParseContext();
                     PercolatorFieldMapper.createQueryBuilderField(version,
                         fieldMapper, queryBuilders[i], parseContext);
-                    indexWriter.addDocument(document);
+                    indexWriter.addDocument(doc);
```

Comment:
This should be `parseContext.doc()` I think? You don't need a separate LuceneDocument instance here.

Reply:
yes it's a leftover. thanks for catching it